sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/CloudEvent.cs

```diff
@@ -85,6 +85,18 @@ public CloudEvent(string source, string type, BinaryData data, string dataConten
             ExtensionAttributes = new Dictionary<string, object>();
         }
 
+        private CloudEvent(string id, string source, string type)
+        {
+            Argument.AssertNotNull(id, nameof(id));
+            Argument.AssertNotNull(source, nameof(source));
+            Argument.AssertNotNull(type, nameof(type));
+
+            Id = id;
+            Source = source;
+            Type = type;
+            ExtensionAttributes = new Dictionary<string, object>();
+        }
```

Comment:
I made this constructor private because the publishing tests started calling this constructor rather than `CloudEvent(string source, string type, object data)`, but that means that the BinaryData extension method `ToCloudEvent` can't call this constructor, so it won't throw an error if `Id` is null. I wasn't really sure how to approach this problem without changing all of the CloudEvent tests.

Reply:
why does the other constructor not have an Id param?

Reply:
We set the id by default

Reply:
ah ok, I think we were going to remove the extension methods anyway.

Reply:
Nvm I just changed the constructor to accept all of the CloudEvent properties so it wouldn't conflict with the other constructor

Reply:
Oh in that case, should I just remove them now?

Reply:
I think so.

Reply:
That's fine too. We can address it later if you are unblocked.